### PR TITLE
Hide popup with 'leave shared VM' label, after opening popup with confirmation

### PR DESCRIPTION
**Pull request by polzoom**
## Description

Add flag `isModalActive`. According to this flag render or not popup with 'leaved shared VM'(1). Hide this popup(1) on opening popup with confirmation(2). On close or press button 'No' popup (1) will be shown.
alt tag
## Motivation and Context

Fix  for #8867.
## How Has This Been Tested?

[video](https://www.youtube.com/watch?v=s1glS2Q-Kxw&feature=youtu.be)
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by usirin** [comment]
Thank you for the PR @polzoom. Can you please fix the commit message and the PR message please? `half of issue fix` doesn't mean a thing in a repository +90k commit :)

**Comment by polzoom** [comment]
Hi @usirin , I agree with you and have changed messages for PR and commit

**Comment by usirin** [comment]
@polzoom Thank you! I see some debug statements in the code, is it still `wip`?

**Comment by polzoom** [comment]
@usirin , yes) Sorry, i forgot to rm `console.log` before commit. If it is necessary, i will rm it

File: client/app/lib/components/sidebarmachineslistitem/leavesharedmachinewidget.coffee [hunk 1]
```diff
@@ -11,42 +11,57 @@ module.exports = class LeaveSharedMachineWidget extends React.Component
   @defaultProps =
     className : 'Approved'
 
+  constructor: (props) ->
+
+    super
+
+    @state = {
```
Comment (hakankaradis) [inline client/app/lib/components/sidebarmachineslistitem/leavesharedmachinewidget.coffee]: one line is enough. `@state = { isModalActive: yes }`

File: client/app/lib/components/sidebarmachineslistitem/leavesharedmachinewidget.coffee [hunk 2]
```diff
@@ -11,42 +11,57 @@ module.exports = class LeaveSharedMachineWidget extends React.Component
...
+      isModalActive : true
+    }
 
   onLeaveClicked: ->
 
+    console.log('@state ', @state)
```
Comment (hakankaradis) [inline client/app/lib/components/sidebarmachineslistitem/leavesharedmachinewidget.coffee]: remove.

File: client/app/lib/components/sidebarmachineslistitem/leavesharedmachinewidget.coffee [hunk 3]
```diff
@@ -11,42 +11,57 @@ module.exports = class LeaveSharedMachineWidget extends React.Component
...
+    @setState {isModalActive: false}
```
Comment (hakankaradis) [inline client/app/lib/components/sidebarmachineslistitem/leavesharedmachinewidget.coffee]: here is the [coffee-styleguide](https://github.com/koding/styleguide-coffeescript) 
`obj = { foo: bar, baz: qux }`

Apply it for all other wrong usage.

**Review by hakankaradis — COMMENTED** [review]
This pr doesn't full solve the problem.
After the leave shared widget closed, you have to click twice to open it back. It is not the way we want.

https://github.com/koding/koding/blob/master/client/app/lib/components/sidebarmachineslistitem/index.coffee#L168

Here `activeLeavingSharedMachineId` came null. 
You need to check if the active machine id is the machine that we are sharing.

Basically, update that line with this
`return null  if @state.activeMachine isnt @machine('_id')`

**Comment by polzoom** [comment]
Hello @hakankaradis Thanks for your review, i have changed it

**Review by hakankaradis — APPROVED** [review]

**Pull request merged**